packages/components/src/components/file-uploader/_file-uploader.scss

```diff
@@ -54,85 +101,165 @@
     margin-bottom: $carbon--spacing-05;
   }
 
-  .#{$prefix}--file-container {
-    display: block;
-    width: 100%;
+  .#{$prefix}--file-btn ~ .#{$prefix}--file-container {
     margin-top: $carbon--spacing-06;
   }
 
   .#{$prefix}--file__selected-file {
-    display: flex;
+    display: grid;
+    grid-gap: $carbon--spacing-03 $carbon--spacing-05;
+    grid-template-columns: 1fr auto;
+    grid-auto-rows: auto;
     align-items: center;
-    justify-content: space-between;
-    height: rem(40px);
-    max-width: rem(300px);
+    min-height: rem(40px);
+    max-width: rem(320px);
     margin-bottom: $carbon--spacing-03;
-    padding: 0 $carbon--spacing-03 0 $carbon--spacing-05;
+    padding: $carbon--spacing-04 0;
     background-color: $field-01;
-    overflow: hidden;
+    word-break: break-word;
 
     &:last-child {
       margin-bottom: 0;
     }
 
-    .#{$prefix}--inline-loading__animation,
+    .#{$prefix}--form-requirement {
+      grid-column-start: 1;
+      grid-column-end: -1;
+      max-height: none;
+      margin: 0;
+    }
+
     .#{$prefix}--loading {
-      right: -0.25rem; // offset for loading svg container
-      width: 1.5rem;
-      height: 1.5rem;
+      width: $carbon--spacing-06;
+      height: $carbon--spacing-06;
+    }
+
+    .#{$prefix}--file-filename {
+      @include type-style('body-short-01');
+      margin-left: $carbon--spacing-05;
+      white-space: nowrap;
+      text-overflow: ellipsis;
+      overflow: hidden;
     }
   }
 
+  // TODO: deprecate this block
+  .#{$prefix}--file__selected-file--invalid__wrapper {
+    @include focus-outline('invalid');
+    outline-width: 1px;
+    background-color: $field-01;
+    max-width: rem(320px);
+    margin-bottom: $carbon--spacing-03;
+  }
+
   .#{$prefix}--file__selected-file--invalid {
     @include focus-outline('invalid');
-    margin-bottom: $carbon--spacing-02;
+    outline-width: 1px;
+  }
+
+  .#{$prefix}--file__selected-file--invalid .#{$prefix}--form-requirement {
+    border-top: 1px solid $ui-03;
+    padding-top: $carbon--spacing-03;
+  }
+
+  .#{$prefix}--file__selected-file--invalid
+    .#{$prefix}--form-requirement__title {
+    padding-right: $carbon--spacing-05;
+    padding-left: $carbon--spacing-05;
+    font-size: rem(12px);
+  }
+
+  .#{$prefix}--file__selected-file--invalid
+    .#{$prefix}--form-requirement__supplement {
+    padding-right: $carbon--spacing-05;
+    padding-left: $carbon--spacing-05;
+    font-size: rem(12px);
```

Comment:
@shixiedesign shouldn't we be using the `label-01` type token rather than just setting the font size to 12px?

> This is for field labels in components and error messages.

https://www.carbondesignsystem.com/guidelines/typography/productive

Reply:
Yep that's right. This is my fault I realize I didn't spec any of the type! @emyarod  Here are the type tokens in this component: 

![image](https://user-images.githubusercontent.com/15144993/64286655-90b38580-cf23-11e9-9702-71425d1b4b91.png)